doc/features/open-telemetry/tracing-rfc.md

```diff
@@ -0,0 +1,414 @@
+
+Similar to the existent metrics feature, this functionality will include a project named `Cassandra.OpenTelemetry` that will extend the core `Cassandra` project and will handle the spans'generation.\
+`Cassandra.OpenTelemetry` has a single dependency from the package `System.Diagnostics.DiagnosticSource` which has the lowest non-deprecated version as [`6.0.1`](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/6.0.1#dependencies-body-tab). Using this version it is possible to target the same .NET frameworks as the `Cassandra.AppMetrics` project which are `netstandard2.0;net461`.
+
+**Note:** There is an alternative that uses `OpenTelemetry.Api` package to avoid code duplication, but that implies changing the minimal target framework from `net461` to `net462`. Such alternative and its drawbacks are mentioned in the section [*Rationale and Alternatives*](#rationale-and-alternatives).
+
+### Extension methods
+
+The project will include a `Builder` extension method named `AddOpenTelemetryInstrumentation` that will instantiate a new class named `Trace` which will start and populate the `System.Diagnostics.Activity` class that will have the Cassandra telemetry information.
+This new class will implement an `IDriverTrace` interface that will be included in the Cassandra core, which is a similar implementation that already exists in the `Cassandra.AppMetrics` package.
+
+## Cassandra core project
+
+### Tracer observers
+
+The tracing implementation will use the already defined factory interfaces for the creation of observer factories, so the following classes will be created:
+
+```csharp
+internal class TracerObserverFactoryBuilder : IObserverFactoryBuilder
+```
+
+```csharp
+internal class TracerObserverFactory : IObserverFactory
+```
+
+When instantiating the factories that will culminate in the creation if the `TracesRequestObserver`, the implementation of `IDriverTrace` will be passed as parameter. This interface will define the following methods:
+
+```csharp
+public interface IDriverTrace
+{
+    void OnStart(IStatement statement);
+
+    void OnError(Exception ex);
+}
+```
+
+and will be used in the `TracesRequestObserver`:
+
+```csharp
+internal class TracesRequestObserver : IRequestObserver
+
+(...)
+
+public void OnRequestError(Host host, RequestErrorType errorType, RetryDecision.RetryDecisionType decision)
+{
+}
+
+public void OnRequestFinish(Exception exception)
+{
+    _driverTrace.OnError(exception);
+}
+
+public void OnRequestStart(IStatement statement)
+{
+    _driverTrace.OnStart(statement);
+}
+
+public void OnSpeculativeExecution(Host host, long delay)
+{
+}
+```
+
+### Public API
+
+The `Builder` will include a new method named `WithTracer` that will include an `IDriverTrace` instance being passed as parameter. This method can be used by anyone to pass their tracer implementation and will be used by the extension method `AddOpenTelemetryInstrumentation()` mentioned in the previous section.
```

Comment:
What is the use case for someone to provide their own `IDriverTrace` implementation?

Reply:
I would say, if someone wants to implement their own instrumentation like not using OTEL or prefer to create their own activity with their own attributes, it would make sense.
Also, using IDriverTracer is a way to separate the interface from its implementation without the use of a provider builder, just like Cassandra.AppMetrics implementation has

Reply:
Ok 👍